be/src/vec/olap/vcollect_iterator.cpp

```
@@ -55,7 +55,8 @@ void VCollectIterator::build_heap(std::vector<RowsetReaderSharedPtr>& rs_readers
         DCHECK(!rs_readers.empty());
         for (auto [c_iter, r_iter] = std::pair {_children.begin(), rs_readers.begin()};
              c_iter != _children.end();) {
-            if ((*c_iter)->init() != Status::OK()) {
+            auto s = (*c_iter)->init();
+            if (!s.ok() && s.precise_code() != OLAP_ERR_DATA_EOF) {
                 delete (*c_iter);
```

Need to return an error status here?

---

Done